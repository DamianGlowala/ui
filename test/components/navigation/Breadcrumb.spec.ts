import { describe, it, expect } from 'vitest'
import { UBreadcrumb } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'
import uiBreadcrumb from '../../../src/runtime/ui.config/navigation/breadcrumb'

describe('Breadcrumb', () => {
  it.each([
    ['basic case', {}],
    ['with links', { props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }], ui: uiBreadcrumb } }],
    ['with divider', { props: { divider: '|', links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }], ui: uiBreadcrumb } }],
    ['with icon', { props: { links: [{ label: 'Home', to: '/', icon: 'i-heroicons-academic-cap' }, { label: 'About', to: '/about', icon: 'i-heroicons-code-bracket-square' }], ui: uiBreadcrumb } }],
    ['with icon slot', { slots: { icon: 'icon slot' }, props: { links: [{ label: 'Home', to: '/', icon: 'i-heroicons-academic-cap' }, { label: 'About', to: '/about', icon: 'i-heroicons-code-bracket-square' }], ui: uiBreadcrumb } }],
    ['with divider slot', { slots: { divider: 'divider slot' }, props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }], ui: uiBreadcrumb } }],
    ['with default slot', { slots: { default: 'default slot' }, props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }], ui: uiBreadcrumb } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Breadcrumb.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UBreadcrumb)
    expect(html).toMatchSnapshot()
  })
})
