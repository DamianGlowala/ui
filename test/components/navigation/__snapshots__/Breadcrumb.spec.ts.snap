// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > renders basic case correctly 1`] = `
"<nav aria-label="Breadcrumb">
  <ol class=""></ol>
</nav>"
`;

exports[`Breadcrumb > renders with divider correctly 1`] = `
"<nav aria-label="Breadcrumb">
  <ol class="">
    <li class=""><a class="" href="/">
        <!--v-if--><span class="">Home</span>
      </a><span role="presentation">|</span></li>
    <li class=""><a class="" aria-current="page" href="/about">
        <!--v-if--><span class="">About</span>
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with divider slot correctly 1`] = `
"<nav aria-label="Breadcrumb">
  <ol class="">
    <li class=""><a class="" href="/">
        <!--v-if--><span class="">Home</span>
      </a>Divider slot</li>
    <li class=""><a class="" aria-current="page" href="/about">
        <!--v-if--><span class="">About</span>
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;
